website/source/docs/platform/k8s/mesh-gateways.html.md

```diff
@@ -0,0 +1,360 @@
+        {
+          "primary_datacenter": "dc1"
+        }
+    server:
+      extraConfig: |
+        {
+          "primary_datacenter": "dc1"
+        }
+    ```
+1. If using Kubernetes, Connect injection must be enabled. This can be set via the helm chart:
+
+    ```yaml
+    connectInject:
+      enabled: true
+    ```
+1. gRPC must be enabled. This can be set via the helm chart:
+
+    ```yaml
+    client:
+      grpc: true
+    ```
+1. If setting a global default [gateway mode](/docs/connect/mesh_gateway.html#modes-of-operation),
+   central config must be enabled for all agents so that they actually use the default mode.
+   This can be set via the helm chart:
+
+    ```yaml
+    connectInject:
+      centralConfig:
+       enabled: true
+    ```
+
+## Guide to Using Mesh Gateways To Connect Two Kubernetes Clusters
+This guide shows how to configure the consul-helm chart to set up gateways
+that route Connect traffic between two Kubernetes clusters each running their
+own Consul datacenter.
+
+### Prequisites
+The prerequisites [specified above](/docs/platform/k8s/mesh-gateways.html#prequisites)
+must be met.
+
+### Step 1 - Get v0.9.0-beta1
+Since Gateways are still in Beta, you'll need a specific consul-helm version.
+
+```sh
+# Clone the chart repo
+git clone https://github.com/hashicorp/consul-helm.git
+cd consul-helm
+
+# Checkout the v0.9.0-beta1 release.
+git checkout v0.9.0-beta1
+
+#  You should see:
+#  Note: checking out 'v0.9.0-beta1'.
+#
+#  You are in 'detached HEAD' state. You can look around, make experimental
+#  ...
+```
+
+### Step 2 - Configure Your First DC
+Add the following config to your helm chart values for dc1.
+
+```yaml
+# dc1-values.yaml
+meshGateway:
+  enabled: true
+
+  # The wan address is what remote datacenters will send their gateway
+  # traffic to.
+  wanAddress:
+    # For now we're setting this to the node IP but in the next step we'll set
+    # it to our Load Balancer's IP.
+    useNodeIP: true
+
+  # When there are no Connect services running, the gateway fails health checks.
+  # As a workaround we will temporarily disable them.
+  enableHealthChecks: false
+
+  # Here we're configuring a LoadBalancer service to front our gateway pods.
+  service:
+    enabled: true
+    type: LoadBalancer
+
+# You should also have the config specified in the prerequisites.
+```
+
+And then upgrade your chart:
+
+```bash
+# Get your chart name.
+helm list
+
+# Then upgrade it.
+helm upgrade <your release name> . -f dc1-values.yaml
+```
+
+You should see the gateway pods starting:
+
+```bash
+$ kubectl get pod -l component=mesh-gateway
+sanguine-elk-consul-mesh-gateway-5c9475b97f-h2g9g                 1/1     Running   0          11m
+sanguine-elk-consul-mesh-gateway-5c9475b97f-ptkbd                 1/1     Running   0          11m
+```
+
+### Step 3 - Wait for the Load Balancer IP
+We're going to use the load balancer's IP to route to the gateways so we need
+to wait for that to be up.
+
+Run
```

Comment:
```suggestion
Search for the load balancer service. 
```

Reply:
The relationship between the loadbalancer and the mesh gateway feels a bit squishy here. I think we're looking for . the loadbalancer that sits in front of the gateway, but it might be that the gateway is the loadbalancer.

Reply:
Yeah it's a bit confusing because often a loadbalancer and a gateway are the same thing. In our case, the gateway is just an app we're running in Kubernetes and since other dc's need to route to it, we need a way to expose it externally... hence the load balancer. I think I can make it clearer though.